packages/dcos-integration-test/extra/test_networking.py

```diff
@@ -281,3 +281,187 @@ def thread_request():
     assert len(set(expected_uuids)) == numapps
     assert len(set(received_uuids)) == numapps
     assert set(expected_uuids) == set(received_uuids)
+
+
+@pytest.mark.skipif(not lb_enabled(), reason='Load Balancer disabled')
+def test_dcos_cni_l4lb(dcos_api_session):
+    '''
+    This tests the `dcos - l4lb` CNI plugins:
+        https: // github.com / dcos / dcos - cni / tree / master / cmd / l4lb
+
+    The `dcos-l4lb` CNI plugins allows containers running on networks that don't
+    necessarily have routes to spartan interfaces and minuteman VIPs to consume DNS
+    service from spartan and layer-4 load-balancing services from minuteman by
+    injecting spartan and minuteman services into the container's network
+    namespace. You can read more about the motivation for this CNI plugin and type
+    of problems it solves in this design doc:
+
+    https://docs.google.com/document/d/1xxvkFknC56hF-EcDmZ9tzKsGiZdGKBUPfrPKYs85j1k/edit?usp=sharing
+
+    In order to test `dcos-l4lb` CNI plugin we emulate a virtual network that lacks
+    routes for spartan interface and minuteman VIPs. In this test, we first install
+    a virtual network called `spartan-net` on one of the agents. The `spartan-net`
+    is a CNI network that is a simple BRIDGE network with the caveat that it
+    doesn't have any default routes. `spartan-net` has routes only for the agent
+    network and the DC/OS overlay (9.x.x.x/8) In other words it doesn't have any
+    routes towards the spartan-interfaces or minuteman VIPs.
+
+    We then run a server (our python ping-pong server) on the DC/OS overlay.
+    Finally to test that the `dcos-l4lb` plugin, which is also part of
+    `spartan-net` is able to inject the Minuteman and Spartan services into the
+    contianer's netns, we start a client on the `spartan-net` and try to `curl` the
+    `ping-pong` server using its VIP. Without the Minuteman and Spartan services
+    injected in the container's netns the expectation would be that this `curl`
+    would fail, with a successful `curl` execution on the VIP allowing the
+    test-case to PASS.
+    '''
+    # Get the first agent. We will run our apps on this specific agent.
+    log.debug("Retrieving agent network")
+    agent_network = None
+    try:
+        agent_network = ipaddress.IPv4Interface(dcos_api_session.slave_list[1] + '/8').network
+    except Exception as ex:
+        raise AssertionError(
+            "Couldn't retrieve the agent network':{}".format(agent_network)) from ex
+
+    log.info("Agent network is:{}".format(agent_network))
+
+    '''
+    Test the l4lb CNI plugin
+    * create an app that will install a CNI configuration for `spartan-net` network on all the agents.
```

Comment:
A minor correction, instead of `on all the agents` should it be `on one of the agents`?